### PR TITLE
Add a self timeout module

### PR DESCRIPTION
**Pull request by ALazyMeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable
- [x] I have tested all changes

Added a selftimeout module, allowing users to timeout themselves for a random amount of time

Thanks @tetyys for the inspiration

### CODE REVIEW

**Comment by TETYYS** [comment]
> using .timeout forsen 20160m literally doesn't work for some reason

I experienced that too when using just `123s`, maybe they removed support for s/m/h/w?

**Comment by ALazyMeme** [comment]
> maybe they removed support for s/m/h/w?

Don't think so, since `.timeout forsen 1w` works just fine. Maybe converting such a high value is too much work for Twitch's servers Kappa

File: pajbot/modules/basic/roll.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                self.bot.send_message_to_user(
+                    source, f"You rolled a {rolled_value}! {self.settings['zero_response']}", event, method="reply"
+                )
+            else:
+                timeout_length = self.seconds_conversion(rolled_value)
+                # Check if timeout value is over Twitch's maximum
+                if timeout_length > 1209600:
+                    timeout_length = 1209600
+
+                self.bot.timeout(source, timeout_length, f"You rolled a {rolled_value}!")
```
Comment (TETYYS) [inline pajbot/modules/basic/roll.py]: check for mod/broadcaster

File: pajbot/modules/basic/roll.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+from pajbot.modules import BaseModule, ModuleSetting
+from pajbot.modules.basic import BasicCommandsModule
+
+log = logging.getLogger(__name__)
+
+
+class RollModule(BaseModule):
+    ID = __name__.split(".")[-1]
+    NAME = "Roll"
+    DESCRIPTION = "Allows users to roll a random number, which can also be used as timeout times"
```
Comment (Felanbird) [inline pajbot/modules/basic/roll.py:13]:
```suggestion
    DESCRIPTION = "Allows users to roll a random number, which can also be used as timeout durations"
```

File: pajbot/modules/basic/roll.py (hunk 1 above)
Comment (ALazyMeme) [reply pajbot/modules/basic/roll.py]: I was under the impression that pajbot automatically checked for this in the timeout function, but I can't seem to find it anywhere. Will have to talk to pajlada about why that is.

**Comment by pajlada** [comment]
Module bloated, strip it down to do what it's actually meant to do (which is time yourself out a random amount of time).
Don't use word "roll" if it's not in the content of dice.

**Review by pajlada — APPROVED** [review]

**Pull request merged**